SignalModelInterpolation/create_signal_ws_new_cat_2d_res_bkg.py

```diff
@@ -68,34 +69,47 @@ def makeWorkspace(models, systematicss, year, cat, workspace_output, mgg_range,
 
   if doSyst:
     systematics = systematicss[year][cat]
-
     #creates splines for const values
-    const_sys_names = [name for name in systematics[masses[0]].keys() if "const" in name]
+    const_sys_names = ["fnuf_mean","fnuf_rate","fnuf_sigma","material_mean","material_rate","material_sigma","MCScale_scale_mean","MCScale_scale_rate","MCScale_scale_sigma","MCSmear_smear_mean","MCSmear_smear_rate","MCSmear_smear_sigma"]
     consts_splines = {}
+
     for systematic in const_sys_names:
-      values = np.asarray([systematics[m][systematic] for m in masses])
+      all_values = []
+      for m in masses:
+        # Check if systematics for mass m is "no systematics" or the systematic key does not exist
+        if systematics[m] == "no systematics" or systematic not in systematics[m]:
+            # Append 0 if no systematics or key doesn't exist
+            all_values.append(0)
+        else:
+            # Append the value from the dictionary
+            all_values.append(systematics[m][systematic])
+
+      values=np.asarray(all_values)
+      #values = np.asarray([systematics[m][systematic] if systematics[m] != "no systematics" else 0 for m in masses])
```

Comment:
I guess this is how we dealt with "no systematics" before? What was the reason to change? Didn't it do the same thing?

Reply:
The commented out line is substituted by the previous 5 lines, i.e. the "for" loop. It's a little different to deal with the "no systematics" term in the json file between resonant background and signals as explained before. Remove the commented line because it's not runnable in this case